py-polars/polars/config.py

```diff
@@ -140,6 +141,7 @@ class ConfigParameters(TypedDict, total=False):
     set_trim_decimal_zeros: bool | None
     set_verbose: bool | None
     set_expr_depth_warning: int
+    set_engine_affinity: Literal["auto", "gpu", "streaming"] | None
```

Comment:
I think this should use the existing `EngineType`. Less maintenance and less confusion I would say.